### PR TITLE
SI-8332 implicit class param unquoting in quasiquotes

### PR DESCRIPTION
**Pull request by densh**
A new api that simplifies handling of implicit parameters has been
mistakingly supporting just methods parameters. This commit adds
support for class parameters too.

review @retronym

### CODE REVIEW

File: src/reflect/scala/reflect/api/Internals.scala [hunk 1]
```diff
@@ -581,7 +581,7 @@ trait Internals { self: Universe =>
     val ImplicitParams: ImplicitParamsExtractor
 
     trait ImplicitParamsExtractor {
-      def apply(paramss: List[List[ValDef]], implparams: List[ValDef]): List[List[ValDef]]
+      def apply(paramss: List[List[Tree]], implparams: List[Tree]): List[List[Tree]]
```
Comment (densh) [inline src/reflect/scala/reflect/api/Internals.scala:584]: API here was accidentally strongly typed and inconsistent with usual DefDef.

**Comment by xeno-by** [comment]
Needs to be rebased.

File: src/compiler/scala/tools/reflect/quasiquotes/Reifiers.scala [hunk 2]
```diff
@@ -151,9 +151,15 @@ trait Reifiers { self: Quasiquotes =>
         mirrorCall(nme.This, tree)
       case SyntacticTraitDef(mods, name, tparams, earlyDefs, parents, selfdef, body) =>
         reifyBuildCall(nme.SyntacticTraitDef, mods, name, tparams, earlyDefs, parents, selfdef, body)
-      case SyntacticClassDef(mods, name, tparams, constrmods, vparamss, earlyDefs, parents, selfdef, body) =>
-        reifyBuildCall(nme.SyntacticClassDef, mods, name, tparams, constrmods, vparamss,
-                                              earlyDefs, parents, selfdef, body)
+      case SyntacticClassDef(mods, name, tparams, constrmods, build.ImplicitParams(vparamss, implparams),
+                             earlyDefs, parents, selfdef, body) =>
+        if (implparams.nonEmpty)
```
Comment (xeno-by) [inline src/compiler/scala/tools/reflect/quasiquotes/Reifiers.scala]: Would it be possible to unify these branches?

Comment (densh) [inline src/compiler/scala/tools/reflect/quasiquotes/Reifiers.scala]: Fixed

**Comment by xeno-by** [comment]
LGTM

**Pull request merged**